api/server/router/container/container_routes.go

```diff
@@ -562,13 +562,20 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 			config.Healthcheck.StartInterval = 0
 		}
 
-		for _, m := range hostConfig.Mounts {
-			if m.BindOptions != nil {
-				// Ignore ReadOnlyNonRecursive because it was added in API 1.44.
-				m.BindOptions.ReadOnlyNonRecursive = false
-				if m.BindOptions.ReadOnlyForceRecursive {
+		for idx, m := range hostConfig.Mounts {
+			if m.Type == mount.TypeBind {
+				bo := m.BindOptions
+				if bo != nil && bo.ReadOnlyForceRecursive {
 					return errdefs.InvalidParameter(errors.New("BindOptions.ReadOnlyForceRecursive needs API v1.44 or newer"))
+				} else {
+					bo = &mount.BindOptions{}
+					hostConfig.Mounts[idx].BindOptions = bo
```

Comment:
Note: This is needed, because `m.BindOptions` would only overwrite the `BindOptions` in the `m` variable which is a local copy of the original mount.